nomad/structs/workload_id.go

```diff
@@ -222,10 +222,18 @@ type AllocIdentitiesResponse struct {
 	QueryMeta
 }
 
+type WorkloadType int
+
+const (
+	ServiceWorkload WorkloadType = iota
+	TaskWorkload
```

Comment:
Existing default identities are `TaskWorkload`, so swapping the order so that the existing value is the zero-value here makes it slightly less likely for us to mess this up across upgrades.

```suggestion
	TaskWorkload
	ServiceWorkload WorkloadType = iota
```